src/docs/release/breaking-changes/services-scheduler-dependency-reversed.md

```diff
@@ -0,0 +1,74 @@
+---
+title: Reversing the dependency between the scheduler and services layer
+description: The services layer now depends on the scheduler layer.
+---
+
+## Summary
+
+The dependency chain between the services layer and the scheduler layer has been
+reversed: The services layer now depends on the scheduler layer. This change may
+affect you if you have defined your own custom binding based on Flutter's
+`SchedulerBinding` and `ServicesBinding`.
+
+## Context
+
+Prior to this change, the scheduler layer was dependent on the services layer.
+This change reverses the dependency chain and allows the services layer to make
+use of the scheduling primitives in the scheduler layer. For example, services
+in the services layer can now schedule tasks via `SchedulerBinding.scheduleTask`.
+
+
+## Description of change
+
+The change only affects users who are defining their own custom bindings based
+on Flutter's `SchedulerBinding` and `ServicesBinding`.
+
+## Migration guide
+
+Prior to this change, the `ServiceBinding` had to be defined before the
+`SchedulerBinding`. With this change, it is the other way around:
+
+Code before migration:
+
+<!-- skip -->
```

Comment:
can we avoid skipping these?

Reply:
These actually probably could be analyzed since they're valid in a single Dart file.

Basically the extract tool rips each of these out into a file with all the `package:flutter` libraries available (and `// ignore_for_file: unused_import`), and then runs the analyzer on that file.